content/ui-patterns/overlays.mdx

```diff
@@ -0,0 +1,450 @@
+
+</div>
+</Grid>
+
+If the height is set to `auto`, make sure the content region has a `max-height` property defined to support overflow scrolling. For overlays that list user-generated information in a single line, use the `large` height for reference.
+
+Avoid creating overlays that are too short or too tall compared to their width. As a general guideline, keep a maximum size ratio of 1.6:1, which translates to the following ranges:
+
+&ZeroWidthSpace;<span style="white-space: nowrap"><sub>Height</sub> ∖ <sup>Width</sup></span> | `small` | `medium` | `large` | `xlarge` | `xxlarge` |
+----------|:-----:|:------:|:-----:|:-----------:|:-----------------:|
+`xsmall`  | ✅    | ✅     | ❌     | ❌          | ❌                 |
+`small`   | ✅    | ✅     | ❌     | ❌          | ❌                 |
+`medium`  | ✅    | ✅     | ✅     | ❌          | ❌                 |
+`large`   | ❌    | ✅     | ✅     | ✅          | ❌                 |
+`xlarge`  | ❌    | ❌     | ✅     | ✅          | ✅                 |
+
+
+## Behaviors
+
+### Opening
+
+Overlays appear on the page after a direct user interaction, such as on click, tap, press, etc.
+
+<!--
+The following paragraph needs a revision based on
+https://github.com/github/primer/issues/575
+
+On desktops and other devices with a fine pointer, overlays may also be triggered by a delayed `hover` event (GitHub calls these _hovercards_). Since this behavior doesn't work on touch devices, consider whether tapping on the source element should open the overlay or take the user to its destination directly. For example, a delayed hover on a **@username** may present a _hovercard_ on desktop, while a tap on the username would open a bottom sheet with a link to the user's profile page.
+-->
+
+
+
+Don't show overlays on page load. For that, prefer in-page messages and flash banners.
+
+<!--
+### Loading
+Pending.
+-->
+
+### Dismissing
+
+Users should be able to dismiss an overlay by:
+- Pressing Escape
+- Pressing on a **Close** button
+- Clicking or tapping outside
+- Pressing the browser’s back button
+
+Close buttons are useful as an easy escape route, particularly in larger or more complex overlays, or when the overlay is no longer visually connected to the source that triggered it. Always use close buttons overlays with backdrops.
+
+Users may click or tap outside to dismiss an overlay. Larger overlays with backdrops may opt to disable this functionality in dangerous scenarios where the user could lose progress or data accidentally.
+
+If the overlay contains changes that are not immediately applied (such as in a form), offer footer buttons such as **Save**, **Apply**, or **Submit**. To avoid users from losing important unsaved changes, a [`beforeUnload`](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload) event may be used.
+
+Users should also be able to click or tap the browser’s back button to dismiss the overlay. This is particularly useful on mobile devices.
+
+
+## Placement
```

Comment:
Nitpick: this section is about placement, but also goes into some more nuanced behavioral specs. I made some comments on how to improve those already, but if you wanted to streamline this even more you could _just_ focus on placement alone here.